File: LC 463. Island Perimeter.cpp
class Solution {
public:
    int islandPerimeter(vector<vector<int>>& grid) {
        int ret = 0, row = grid.size(), col = grid[0].size();
        vector<vector<int>> visited = grid;
        queue<int> q1, q2;
        for(int i=0; i<row; i++){
            for(int j=0; j<col; j++){
                if(grid[i][j] == 1){
                    ret += 4;
                    if(i > 0){
                        ret -= grid[i-1][j];
                    }
                    if(i<row-1){
                        ret -= grid[i+1][j];
                    }
                    if(j > 0){
                        ret -= grid[i][j-1];
                    }
                    if(j < col-1){
                        ret -= grid[i][j+1];
                    }
                }
            }
        }
        return ret;
    }
};

File: LC 463. Island Perimeter.java
class Solution {
    public int islandPerimeter(int[][] grid) {
        int m = grid.length, n = grid[0].length, ret = 0;
        for(int i=0; i<m; i++){
            for(int j=0; j<n; j++){
                if(grid[i][j] == 1){
                    if(i == 0 || grid[i-1][j] == 0){
                        ret++;
                    }
                    if(j == 0 || grid[i][j-1] == 0){
                        ret++;
                    }
                    if(i == m-1 || grid[i+1][j] == 0){
                        ret++;
                    }
                    if(j == n-1 || grid[i][j+1] == 0){
                        ret++;
                    }
                }
            }
        }
        return ret;
    }
}
